tgui/packages/tgui/interfaces/Manufacturer/index.tsx

```diff
@@ -0,0 +1,321 @@
+/**
+ * @file
+ * @copyright 2024
+ * @author Romayne (https://github.com/MeggalBozale)
+ * @license ISC
+ */
+
+import { useBackend, useLocalState, useSharedState } from '../../backend';
+import { Window } from '../../layouts';
+import { toTitleCase } from 'common/string';
+import { Button, Collapsible, Divider, Input, LabeledList, ProgressBar, Section, Slider, Stack } from '../../components';
+import { formatMoney } from '../../format';
+import { MaintenencePanel, Manufacturable, ManufacturerData, Ore, QueueBlueprint, Resource, Rockbox } from './type';
+
+import { BlueprintButton } from './blueprintButton';
+import { ProductionCard } from './productionCard';
+import { clamp } from 'common/math';
+import { CollapsibleWireMenu } from './collapsibleWireMenu';
+import { pluralize } from '../common/stringUtils';
+
+const CardInfo = (_, context) => {
+  const { data, act } = useBackend<ManufacturerData>(context);
+  return (data.card_owner === null || data.card_balance === null) ? (
+    <Section
+      textAlign="center"
+    >
+      <Stack vertical>
+        <Stack.Item>
+          No Account Found.
+        </Stack.Item>
+        <Stack.Item>
+          <Button icon="add" onClick={() => act("card", { "scan": true })}>Add Account</Button>
+        </Stack.Item>
+      </Stack>
+    </Section>
+  ) : (
+    <Section
+      title="Account Info"
+      buttons={<Button icon="minus" onClick={() => act("card", { "remove": true })}>Log Out</Button>}
+    >
+      <LabeledList>
+        <LabeledList.Item
+          label="Owner"
+        >
+          {data.card_owner}
+        </LabeledList.Item>
+        <LabeledList.Item
+          label="Balance"
+        >
+          {formatMoney(data.card_balance)}⪽
+        </LabeledList.Item>
+      </LabeledList>
+    </Section>
+  );
+};
+
+export const Manufacturer = (_, context) => {
+  const { act, data } = useBackend<ManufacturerData>(context);
+  const [repeat, toggleRepeatVar] = useSharedState(context, "repeat", data.repeat);
+  const [search, setSearchData] = useLocalState(context, "query", "");
+  const [swap, setSwappingMaterial] = useLocalState(context, "swap", null);
+  const wirePanel:MaintenencePanel = {
+    indicators: data.indicators,
+    wires: data.wires,
+    wire_bitflags: data.wire_bitflags,
+  };
+  let toggleRepeat = () => {
+    act("repeat");
+    toggleRepeatVar(!repeat);
+  };
```

Comment:
```suggestion
  let toggleRepeat = () => {
    act("repeat");
    toggleRepeatVar(!repeat);
  };
```
Give that this actually has backend changes, this should be data that is passed back and forth rather than using a second state layer (shared state).

Reply:
AH RIGHT sorry i picked up that idea after the pump control shenanagains you're right